internal/cli/atlas/serverless/backup/restores/watch.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package restores
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type WatchOpts struct {
+	cli.GlobalOpts
+	cli.WatchOpts
+	id          string
+	clusterName string
+	store       store.ServerlessRestoreJobsDescriber
+}
+
+func (opts *WatchOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *WatchOpts) watcher() (bool, error) {
+	result, err := opts.store.ServerlessRestoreJob(opts.ConfigProjectID(), opts.clusterName, opts.id)
+	if err != nil {
+		return false, err
+	}
+	return result.Expired || result.Cancelled || *result.Failed || result.FinishedAt != "", nil
```

Comment:
That does return error as nil right?

Reply:
yes, but if the api call fails it propagates the error on line 49

Reply:
So does this command return 1 if status is `result.Failed` or `result.Expired`

Reply:
well it returns `true, nil`

the boolean is a signal to stop watching

if there is an error it also stops watching but propagates the error as well

Reply:
https://github.com/mongodb/mongodb-atlas-cli/blob/70b59c8bb17f5aeb19231c2d3c51067c5f988cd5/internal/cli/watch_opts.go#L46